AmazonChimeSDK/AmazonChimeSDK/audiovideo/video/backgroundfilter/BackgroundFilterProcessor.swift

```diff
@@ -0,0 +1,236 @@
+                                           (Double(segmentationProcessorHeight)/Double(segmentationProcessorWidth)))
+        // Down sample the image.
+
+        guard let downSampleImage: CIImage = resizeImage(image: inputFrameCI,
+                                                         scale: downSampleScale,
+                                                         aspectRatio: downSampleAspectRatio)
+        else {
+            return nil
+        }
+
+        // Check if segmentation model has loaded.
+        if segmentationProcessor.getModelState() != CwtModelState.LOADED {
+            logger.error(msg: "Segmentation processor failed to start. Unable to perform segmentation.")
+            return nil
+        }
+
+        guard let downSampleImageCg = context.createCGImage(downSampleImage,
+                                                            from: downSampleImage.extent)
+        else {
+            return nil
+        }
+
+        // Convert the input CGImage to a UInt8 byte array.
+        guard var byteArray: [UInt8] = ImageConversionUtils.cgImageToByteArray(cgImage: downSampleImageCg) else {
+            logger.error(msg: "Error converting CGImage to byte array when creating the foreground mask.")
+            return nil
+        }
+
+        // Copy the input buffer to the TensorFlow buffer which will be used during predict.
+        let inputBuffer: UnsafeMutablePointer<UInt8> = segmentationProcessor.getInputBuffer()
+        inputBuffer.initialize(from: &byteArray, count: byteArray.count)
+
+        // Predict the foreground mask.
+        let predictResult: CwtPredictResult = segmentationProcessor.predict()
+        if predictResult != CwtPredictResult.SUCCESS {
+            logger.error(msg: "Error predicting the foreground mask.")
+            return nil
+        }
+
+        // Retrieve the foreground mask.
+        let maskOutputBuffer = segmentationProcessor.getOutputBuffer()
+
+        guard let maskImage: CGImage = ImageConversionUtils.byteArrayToCGImage(raw: maskOutputBuffer,
+                                                                               frameWidth: segmentationProcessorWidth,
+                                                                               frameHeight: segmentationProcessorHeight,
+                                                                               bytesPerPixel: imageChannels,
+                                                                               bitsPerComponent: inputFrameCG.bitsPerComponent)
+        else {
+            logger.error(msg: "Error creating CGImage of the foreground mask.")
+            return nil
+        }
+
+        let maskImageCi = CIImage(cgImage: maskImage)
+
+        // Calculate the scale and aspect ratio factor to upscale.
+        let upSampleScale = Double(inputFrameCG.height) / Double(segmentationProcessorHeight)
+        let upSampleAspectRatio = Double((Double(segmentationProcessorHeight)/Double(segmentationProcessorWidth)) /
```

Comment:
hmmm just curious can't we just do `upSampleAspectRatio  = 1 / downSampleAspectRatio`?